packages/input/src/index.vue

```diff
@@ -0,0 +1,471 @@
+    const inputSize = computed(() => {
+      return props.size || _elFormItemSize.value || (ELEMENT || {}).size
+    })
+    const needStatusIcon = computed(() => {
+      return elForm ? elForm.statusIcon : false
+    })
+    const validateState = computed(() => {
+      return elFormItem ? elFormItem.validateState : ''
+    })
+    const validateIcon = computed(() => {
+      return {
+        validating: 'el-icon-loading',
+        success: 'el-icon-circle-check',
+        error: 'el-icon-circle-close',
+      }[validateState.value]
+    })
+    const textareaStyle = computed(() => {
+      return Object.assign({}, _textareaCalcStyle.value, { resize: props.resize })
+    })
+    const inputDisabled = computed(() => {
+      return props.disabled || (elForm || {}).disabled
+    })
+    const nativeInputValue = computed(() => {
+      return String(props.modelValue)
+    })
+    const showClear = computed(() => {
+      return props.clearable &&
+        !inputDisabled.value &&
+        !props.readonly &&
+        nativeInputValue.value &&
+        (focused.value || hovering.value)
+    })
+    const showPwdVisible = computed(() => {
+      return props.showPassword &&
+        !inputDisabled.value &&
+        !props.readonly &&
+        (!!nativeInputValue.value || focused.value)
+    })
+    const isWordLimitVisible = computed(() => {
+      return props.showWordLimit &&
+        ctx.attrs.maxlength &&
+        (props.type === 'text' || props.type === 'textarea') &&
+        !inputDisabled.value &&
+        !props.readonly &&
+        !props.showPassword
+    })
+    const upperLimit = computed(() => {
+      return ctx.attrs.maxlength
+    })
+    const textLength = computed(() => {
+      return typeof props.modelValue === 'number' ? String(props.modelValue).length : (props.modelValue || '').length
+    })
+    const inputExceed = computed(() => {
+      // show exceed style if length of initial value greater then maxlength
+      return isWordLimitVisible.value && (textLength.value > upperLimit.value)
+    })
+
+    const resizeTextarea = () => {
+      if (isServer) return
+      if (props.type !== 'textarea') return
+      const autosize = props.autosize as AutosizeProp
+      if (!autosize) {
+        _textareaCalcStyle.value = {
+          minHeight: calcTextareaHeight(instance.refs.textarea).minHeight,
+        }
+        return
+      }
+      const minRows = typeof autosize === 'object' ? autosize.minRows : void 0
+      const maxRows = typeof autosize === 'object' ? autosize.maxRows : void 0
+      _textareaCalcStyle.value = calcTextareaHeight(instance.refs.textarea, minRows, maxRows)
+    }
+    const setNativeInputValue = () => {
+      const input = getInput()
+      if (!input) return
+      if (input.value === nativeInputValue.value) return
+      input.value = nativeInputValue.value
+    }
+    const calcIconOffset = place => {
+      let elList = [].slice.call(instance.vnode.el.querySelectorAll(`.el-input__${place}`) || [])
```

Comment:
There are twice references to `instance.vnode.el`, maybe extract it out would be better